starfish/core/segmentation_mask.py

```diff
@@ -101,6 +111,12 @@ def append(self, mask: xr.DataArray):
         validate_segmentation_mask(mask)
         self._masks.append(mask)
 
+        for axis in Axes:
+            if axis.value in mask.coords:
+                max_val = mask.coords[axis.value].values[-1]
+                if self.max_shape[axis] is None or max_val > self.max_shape[axis]:
```

Comment:
```suggestion
                if self.max_shape[axis] is None or max_val >= self.max_shape[axis]:
```

if the max_val is the same as max_shape, i think you need to up max_shape.

Reply:
wait could you explain this more? I don't quite understand

Reply:
So let's say max_vals are (15, 16, 13, 10).  The max shape, in this case, ought to be 17.

However, you would set max_shape to 16 when evaluating max_val=15, and then when you get to max_val=16, you won't bump it to 17.